Expose more of siunitx's capabilities

bjodah opened the pull request:
This extends the capabilities of `Quantity` to pass options to siunitx command `\SI`

I struggled a bit with dealing with escaping of `Options`. Note the suggested change in behaviour
of `Parameters` and the modification of private variable `_escape` in `quantities.py`.
Is there another preferred way of achieving this which I have overlooked?

TODO:
- [x] Allow to pass options
- [x] Support for `quantities.UncertainQuantity`
- [x] Support for using Python `float` with `pylatex.Quantity` (which would map to `siunitx`'s `\num{}`)
- [x] Update `changelog.rst`

JelteF commented:
The suggested `Parameters` change seems fine although I'm not sure it is the best way to check it for being a dict.

JelteF commented:
I've made some comments on your code. Do you want this merged after you fix those or should I wait until you fix the other points (except for the changelog).

bjodah commented:
@JelteF, thank you for your feedback. I changed my code according to your suggestions.
I went ahead and added support for UncertainQuantity and float/int respectively. Test failures look unrelated. Do you want me to update `changelog.rst`?

JelteF commented:
I added two more comments. And yes could you update the changelog?

bjodah commented:
Absolutely, I made the changes you requested. Let me know what you think.

JelteF commented:
Looks good I'll merge it.
